### PR TITLE
fix(module:cascader): fix empty shown when searching value

### LINKED ISSUE #2903: cascader: no result and search result appear at the same time
Developing version

image

### PR DESCRIPTION
**Pull request by wendzhue**
close #2903 

## PR Checklist
Please check if your PR fulfills the following requirements:

- [ ] The commit message follows our guidelines: https://github.com/NG-ZORRO/ng-zorro-antd/blob/master/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

```
[ ] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[ ] Application (the showcase website) / infrastructure changes
[ ] Other... Please describe:
```

## What is the current behavior?

Issue Number: N/A

## Does this PR introduce a breaking change?
```
[ ] Yes
[ ] No
```

### CODE REVIEW

File: components/cascader/nz-cascader.component.html [hunk 1]
```diff
@@ -78,7 +78,7 @@
         (mouseleave)="onOptionMouseLeave(option, i, $event)"
         (click)="onOptionClick(option, i, $event)">
       </li>
-      <li *ngIf="isSearching"
+      <li *ngIf="isSearching && !columns[0].length"
```
Comment (vthinkxie) [inline components/cascader/nz-cascader.component.html]: columns[0] may be null?

Comment (wendzhue) [reply components/cascader/nz-cascader.component.html]: @vthinkxie This is secured here

https://github.com/NG-ZORRO/ng-zorro-antd/blob/07f76af35690417571ca6c32a947f3b7b260ffc6/components/cascader/nz-cascader.component.ts#L715

And I wrote a test about it.

Comment (vthinkxie) [reply components/cascader/nz-cascader.component.html]: https://github.com/NG-ZORRO/ng-zorro-antd/blob/07f76af35690417571ca6c32a947f3b7b260ffc6/components/cascader/nz-cascader.component.ts#L130
should be
`columns: CascaderOption[][] = [ [] ];`
?

Comment (wendzhue) [reply components/cascader/nz-cascader.component.html]: If so there would be an blank dropdown.

Comment (vthinkxie) [reply components/cascader/nz-cascader.component.html]: fine...

**Pull request merged**